internal/vault/credentialproviders/kubernetes.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package credentialproviders
+
+import (
+	"context"
+
+	secretsv1alpha1 "github.com/hashicorp/vault-secrets-operator/api/v1alpha1"
+	authv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type KubernetesCredentialProvider struct {
+	authObj           *secretsv1alpha1.VaultAuth
+	providerNamespace string
+	uid               types.UID
+}
+
+func NewKubernetesCredentialProvider(authObj *secretsv1alpha1.VaultAuth, providerNamespace string,
+	uid types.UID,
+) *KubernetesCredentialProvider {
+	return &KubernetesCredentialProvider{
+		authObj,
+		providerNamespace,
+		uid,
+	}
+}
```

Comment:
I wonder if this function is necessary, since KubernetesCredentialProvider is already public?

Reply:
All fields from `KubernetesCredentialProvider` are currently private. There were no usage of referencing fields when the struct was private, prior to the PR. I also see only [tests](https://github.com/hashicorp/vault-secrets-operator/pull/131/files#diff-43563f2031dc36837155f26a710e12785ebba21517a705364a25bdb08a24cf62) need to instantiate with custom values for the fields. I wonder if adding `SetUID()` would make more sense since only `uid` is used to create the object.

Reply:
Oh yep, sorry, I missed that this was moved to a new package.